libr/io/io_map.c

```diff
@@ -693,7 +703,8 @@ R_API void r_io_map_drain_overlay(RIOMap *map) {
 			if (!r_queue_is_empty (q)) {
 				MapOverlayChunk *start = (MapOverlayChunk *)start_n->data;
 				const ut64 new_size = r_itv_end (cur->itv) - r_itv_begin (start->itv);
-				ut8 *buf = realloc (start->buf, new_size * sizeof (ut8));
+				const st64 new_size2 = new_size * sizeof (ut8);
```

Comment:
realloc expects size_t which is unsigned, so let's avoid passing signed integers to it. realloc (ptr, 0) is UB, since c23. so we'd only need to check for that specific case. i suggest something like this above the original call to realloc:

```
if (!(newsize * sizeof (ut8))) {
   start_n = cur_n = next_n;
   next_n = r_rbnode_next (next_n);
   continue;
}
```

Reply:
the signe dcheck is to detec the overflow in the multiplication. thing is that i didnt though about the useless multiplication, so yeah it should be unsigned indeed

Reply:
so wat du, do you wanna finish this pr?